sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFileWholeTextReader.scala

```diff
@@ -45,6 +45,7 @@ class HadoopFileWholeTextReader(file: PartitionedFile, conf: Configuration)
     val attemptId = new TaskAttemptID(new TaskID(new JobID(), TaskType.MAP, 0), 0)
     val hadoopAttemptContext = new TaskAttemptContextImpl(conf, attemptId)
     val reader = new WholeTextFileRecordReader(fileSplit, hadoopAttemptContext, 0)
+    reader.setConf(hadoopAttemptContext.getConfiguration)
```

Comment:
`WholeTextFileRecordReader` is `Configurable`, `setConf` should be called after creation.
This is why tests are failing before this patch.

However, I am wondering for `org.apache.spark.input.WholeTextFileRecordReader` and `org.apache.spark.input.ConfigurableCombineFileRecordReader`, we can already retrieve config from `org.apache.hadoop.mapreduce.TaskAttemptContext`.  There is no need to make these class `Configurable`

I am wondering if we should remove `Configurable` trait for the related classes all at once.  what do you think @gatorsmile

Reply:
Is there an existing test that fails without this change, as you mention? should it be reenabled?

Reply:
![image](https://user-images.githubusercontent.com/807537/64126084-0c57ea00-cddf-11e9-9701-814ac131e79f.png)
Some tests in `WholeTextFileSuite` and `SaveLoadSuite` are failing without this change.

However, the failure is introduced by my change to `WholeTextFileRecordReader` https://github.com/apache/spark/blob/149de72c220cbc094f0b8756c535cf1bd796a48e/core/src/main/scala/org/apache/spark/input/WholeTextFileRecordReader.scala#L70-L73

We use `getConf` instead of `new Configuration`, then should call `setConf` first.

Reply:
I see, they're not failing in master but can fail if run in an env where Hadoop config files are present?

Reply:
> I see, they're not failing in master but can fail if run in an env where Hadoop config files are present?I see, they're not failing in master but can fail if run in an env where Hadoop config files are present?

Eh, yes, they are not failing in master. The code(master) even normally won't fail in an env where Hadoop configs are present. They could fail or get unexpected result unless the Hadoop configs are incorrectly configured in executor env(such as yarn-cluster), even user supplies correct configs (passed to `TaskAttemptContext`